wizard.py

```diff
@@ -0,0 +1,1038 @@
+# -*- coding: utf-8 -*-
+#!/usr/bin/env python
+"""
+<Program Name>
+  wizard.py
+
+<Author>
+  Lukas Puehringer <lukas.puehringer@nyu.edu>
+
+<Started>
+  April 06, 2017
+
+<Copyright>
+  See LICENSE for licensing information.
+
+<Purpose>
+  Flask web app that provides a wizard to guide through an in-toto
+  software supply chain layout creation.
+
+<Overview of Functions>
+  Utils:
+      Mostly used to transform data
+
+  NoSQL Helpers:
+      The app uses MongoDB to persist user posted data for auditing and to
+      improve the app. NoSQL Helpers are a couple of custom wrappers around
+      common queries.
+
+  View Decorator & Hooks:
+      Currently there is one view decorator for session handling (sessions are
+      used to isolate user posted data) and an after request hook to inject
+      messages from the Flask's message flash framework into ajax responses.
+
+  Views:
+      Each view is an entry point for an HTTP request (c.f. paths in @app.route
+      decorator). Most views correspond to a page the user surfs to and
+      therefor render and return the according template on a GET request.
+      If a page has form postable content, the according view also accepts
+      a POST request which is usually sent when the user clicks NEXT as s/he
+      walks through the pages. The view then handles and persists the posted
+      data and in case everything went well redirects to the text view (page).
+      The views in this document are ordered from top to bottom in the same
+      order as the user walks from page to page.
+      Some pages have additional views associated that accept ajax calls,
+      e.g. to upload or remove files.
+
```

Comment:
It feels odd to have **all** of this in a single file. Why is this?

Reply:
I have the feeling that flask apps often start off like that. At least mine do. Probably because it's a *micro-framework* I also keep the file structure *micro*. In any case, I agree that it makes sense to move some things out. The file is already structured (utils, decorators, view, ...) with that in mind.

Reply:
> “Micro” does not mean that your whole web application has to fit into a single Python file (although it certainly can), nor does it mean that Flask is lacking in functionality.

From their docs. I can see how the rationale was to sprint this and make it work as soon as possible, but I see absolutely no benefit in having everything stuck in just one file.

Reply:
Will fix with https://github.com/in-toto/layout-web-tool/issues/30